server.py

```diff
@@ -0,0 +1,234 @@
+
+
+@app.route("/request-otp", methods=["POST"])
+def request_otp():
+    try:
+        data = request.form
+        all_fields = {
+            "roll_number": data.get("roll_number"),
+            "password": data.get("password"),
+            "secret_answer": data.get("secret_answer"),
+            "sessionToken": request.headers["Session-Token"],
+        }
+        missing = check_missing_fields(all_fields)
+        if len(missing) > 0:
+            return ErpResponse(
+                False, f"Missing Fields: {', '.join(missing)}", status_code=400
+            ).to_response()
+
+        login_details = erp.get_login_details(
+            ROLL_NUMBER=all_fields["roll_number"],
+            PASSWORD=all_fields["password"],
+            secret_answer=all_fields["secret_answer"],
+            sessionToken=all_fields["sessionToken"],
+        )
+
+        session = requests.Session()
+        erp_utils.set_cookie(session, "JSESSIONID", all_fields["sessionToken"])
+        erp.request_otp(
+            headers=headers, session=session, login_details=login_details, log=True
+        )
+
+        return ErpResponse(
+            True, message="OTP has been sent to your connected email accounts"
+        ).to_response()
+    except erp.ErpLoginError as e:
+        return ErpResponse(False, str(e), status_code=401).to_response()
+    except Exception as e:
+        return ErpResponse(False, str(e), status_code=500).to_response()
+
+
+@app.route("/login", methods=["POST"])
+def login():
+    try:
+        data = request.form
+        all_fields = {
+            "roll_number": data.get("roll_number"),
+            "password": data.get("password"),
+            "secret_answer": data.get("secret_answer"),
+            "otp": data.get("otp"),
+            "sessionToken": request.headers["Session-Token"],
+        }
+        missing = check_missing_fields(all_fields)
+        if len(missing) > 0:
+            return ErpResponse(
+                False, f"Missing Fields: {', '.join(missing)}", status_code=400
+            ).to_response()
+
+        login_details = erp.get_login_details(
+            ROLL_NUMBER=all_fields["roll_number"],
+            PASSWORD=all_fields["password"],
+            secret_answer=all_fields["secret_answer"],
+            sessionToken=all_fields["sessionToken"],
+        )
+        login_details["email_otp"] = all_fields["otp"]
+
+        session = requests.Session()
+        erp_utils.set_cookie(session, "JSESSIONID", all_fields["sessionToken"])
+        ssoToken = erp.signin(
+            headers=headers, session=session, login_details=login_details, log=True
+        )
+
+        return ErpResponse(True, data={"ssoToken": ssoToken}).to_response()
+    except erp.ErpLoginError as e:
+        return ErpResponse(False, str(e), status_code=401).to_response()
+    except Exception as e:
+        return ErpResponse(False, str(e), status_code=500).to_response()
+
+
+@app.route("/timetable", methods=["POST"])
+def download_ics():
```

Comment:
How about importing and calling the necessary functions from the cli (gyft.py) rather than having the same logic written twice?